illumos-utils/src/lib.rs

```diff
@@ -90,8 +90,33 @@ mod inner {
         }))
     }
 
-    // Helper function for starting the process and checking the
+    // Helper functions for starting the process and checking the
     // exit code result.
+
+    pub fn spawn(
+        command: &mut std::process::Command,
+    ) -> Result<std::process::Child, ExecutionError> {
+        command.spawn().map_err(|err| ExecutionError::ExecutionStart {
+            command: to_string(command),
+            err,
+        })
+    }
```

Comment:
I think this routine should actually return a newtype that combines the `Command` and the `Child` so that we can pass _that_ to `run_child()` below.  Otherwise, the `run_child()` interface is pretty easy to misuse: nothing structural prevents passing a `Child` that does not match the `Command`.